src/main/telemetry/telemetry.h

```diff
@@ -42,32 +42,55 @@ typedef enum {
 } frskyUnit_e;
 
 typedef enum {
-    SENSOR_VOLTAGE         = 1 << 0,
-    SENSOR_CURRENT         = 1 << 1,
-    SENSOR_FUEL            = 1 << 2,
-    SENSOR_MODE            = 1 << 3,
-    SENSOR_ACC_X           = 1 << 4,
-    SENSOR_ACC_Y           = 1 << 5,
-    SENSOR_ACC_Z           = 1 << 6,
-    SENSOR_PITCH           = 1 << 7,
-    SENSOR_ROLL            = 1 << 8,
-    SENSOR_HEADING         = 1 << 9,
-    SENSOR_ALTITUDE        = 1 << 10,
-    SENSOR_VARIO           = 1 << 11,
-    SENSOR_LAT_LONG        = 1 << 12,
-    SENSOR_GROUND_SPEED    = 1 << 13,
-    SENSOR_DISTANCE        = 1 << 14,
-    ESC_SENSOR_CURRENT     = 1 << 15,
-    ESC_SENSOR_VOLTAGE     = 1 << 16,
-    ESC_SENSOR_RPM         = 1 << 17,
-    ESC_SENSOR_TEMPERATURE = 1 << 18,
-    ESC_SENSOR_ALL         = ESC_SENSOR_CURRENT \
-                            | ESC_SENSOR_VOLTAGE \
-                            | ESC_SENSOR_RPM \
-                            | ESC_SENSOR_TEMPERATURE,
-    SENSOR_ALL             = (1 << 19) - 1,
+    SENSOR_VOLTAGE = 0,
+    SENSOR_CURRENT,
+    SENSOR_FUEL,
+    SENSOR_MODE,
+    SENSOR_ACC_X,
+    SENSOR_ACC_Y,
+    SENSOR_ACC_Z,
+    SENSOR_PITCH,
+    SENSOR_ROLL,
+    SENSOR_HEADING,
+    SENSOR_ALTITUDE,
+    SENSOR_VARIO,
+    SENSOR_LAT_LONG,
+    SENSOR_GROUND_SPEED,
+    SENSOR_DISTANCE,
+    ESC_SENSOR_CURRENT,
+    ESC_SENSOR_VOLTAGE,
+    ESC_SENSOR_RPM,
+    ESC_SENSOR_TEMPERATURE,
+    SENSOR_COUNT    // must be last
 } sensor_e;
 
+typedef enum {
```

Comment:
Could probably be substituted with `#define SENSOR_MASK(x) (1 << x)`.

Reply:
I thought about that but assumed it would:
1. Use more flash
2. Make the `ESC_SENSOR_ALL` variant difficult
```
ESC_SENSOR_ALL_MASK         = ESC_SENSOR_CURRENT_MASK \
	                     | ESC_SENSOR_VOLTAGE_MASK \
	                     | ESC_SENSOR_RPM_MASK \
	                     | ESC_SENSOR_TEMPERATURE_MASK,
```

Also the `SENSOR_ALL` option would be difficult.
```
SENSOR_ALL_MASK             = (1 << SENSOR_COUNT) - 1
```

Reply:
1. It should not use more flash. the constant expression will be evaluated / replaced at compile time, just like it is now;
2.

```
#define ESC_SENSOR_ALL_MASK         = SENSOR_MASK(ESC_SENSOR_CURRENT) \
	                     | SENSOR_MASK(ESC_SENSOR_VOLTAGE) \
	                     | SENSOR_MASK(ESC_SENSOR_RPM) \
	                     | SENSOR_MASK(ESC_SENSOR_TEMPERATURE)

#define SENSOR_ALL_MASK             = (1 << SENSOR_COUNT) - 1
```

Not super nice, but workable. And more maintainable than requiring 2 lists that need to be kept in synch.

Reply:
Ok, I'll revise. As I mentioned above it's going to need a rebase after #7278 anyway.

Reply:
The macro:
```
#define ESC_SENSOR_ALL_MASK = SENSOR_MASK(ESC_SENSOR_CURRENT) \
	                         | SENSOR_MASK(ESC_SENSOR_VOLTAGE) \
	                         | SENSOR_MASK(ESC_SENSOR_RPM) \
	                         | SENSOR_MASK(ESC_SENSOR_TEMPERATURE)
```

Causes problems for the PG initializer because it's not a constant.
```
PG_RESET_TEMPLATE(telemetryConfig_t, telemetryConfig,
    .telemetry_inverted = false,
    .halfDuplex = 1,
    .gpsNoFixLatitude = 0,
    .gpsNoFixLongitude = 0,
    .frsky_coordinate_format = FRSKY_FORMAT_DMS,
    .frsky_unit = FRSKY_UNIT_METRICS,
    .frsky_vfas_precision = 0,
    .hottAlarmSoundInterval = 5,
    .pidValuesAsTelemetry = 0,
    .report_cell_voltage = false,
    .flysky_sensors = {
            IBUS_SENSOR_TYPE_TEMPERATURE,
            IBUS_SENSOR_TYPE_RPM_FLYSKY,
            IBUS_SENSOR_TYPE_EXTERNAL_VOLTAGE
    },
    .disabledSensors = ESC_SENSOR_ALL_MASK,
    .mavlink_mah_as_heading_divisor = 0,
);
```
Any suggested workarounds?

Reply:
I am lost a bit there - the macro is a constant (or there is something I am not seeing).

Reply:
It's defined like this:
```
#define SENSOR_MASK(x) (1 << x)

#define ESC_SENSOR_ALL_MASK = SENSOR_MASK(ESC_SENSOR_CURRENT) \
	                         | SENSOR_MASK(ESC_SENSOR_VOLTAGE) \
	                         | SENSOR_MASK(ESC_SENSOR_RPM) \
	                         | SENSOR_MASK(ESC_SENSOR_TEMPERATURE)
```
And apparently the compiler doesn't like it and chokes on the initializer line.
```
./src/main/telemetry/telemetry.c:63:138: error: initializer element is not constant
 PG_RESET_TEMPLATE(telemetryConfig_t, telemetryConfig,
                                                                                                                                          ^                  
./src/main/telemetry/telemetry.c:63:138: note: (near initialization for 'pgResetTemplate_telemetryConfig.disabledSensors')
```
I tried making it a single line and also wrapping in parens in case it was just a syntax issue from the substitution but no difference. I just think it's unhappy with the function defined by `SENSOR_MASK`.

Reply:
Does it make a difference if you assign explicit values to `ESC_SENSOR_CURRENT`, `ESC_SENSOR_VOLTAGE`, `ESC_SENSOR_RPM`, and `ESC_SENSOR_TEMPERATURE`?

Reply:
It's ugly, but I suppose I can always just add back a single value enum with this like:
```
typedef enum {
    ESC_SENSOR_ALL_MASK = SENSOR_MASK(ESC_SENSOR_CURRENT) \
	                     | SENSOR_MASK(ESC_SENSOR_VOLTAGE) \
	                     | SENSOR_MASK(ESC_SENSOR_RPM) \
	                     | SENSOR_MASK(ESC_SENSOR_TEMPERATURE)
} escSensorAllMask_e;
```
This seems to work and won't add any ongoing maintenance (just confusing).

Reply:
Either way is good. In any case, I think the `ESC_SENSOR_ALL_MASK` macro needs enclosing braces though.

Reply:
There is no = in #define syntax, the = sign is expanded with macro

Reply:
Lol, it's been staring at us all of the time. :stuck_out_tongue_winking_eye: